datahub-gms-graphql-api/src/main/java/com/linkedin/datahub/graphql/api/QueryGraphQLInvocation.java

```diff
@@ -0,0 +1,47 @@
+package com.linkedin.datahub.graphql.api;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.dataloader.DataLoaderRegistry;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.request.WebRequest;
+
+import com.linkedin.datahub.graphql.GraphQLEngine;
+import com.linkedin.datahub.graphql.QueryContext;
+import com.linkedin.datahub.graphql.context.SpringQueryContext;
+
+import graphql.ExecutionInput;
+import graphql.ExecutionResult;
+import graphql.GraphQL;
+import graphql.spring.web.servlet.GraphQLInvocation;
+import graphql.spring.web.servlet.GraphQLInvocationData;
+
+@Component
+@Primary
+public class QueryGraphQLInvocation implements GraphQLInvocation {
+
+    @Autowired
+    GraphQL graphQL;
+
+    @Autowired
+    GraphQLEngine graphQLEngine;
+
+    public static final String APPNAME = "DatahubGmsGraphQLApp";
+
+    @Override
+    public CompletableFuture<ExecutionResult> invoke(GraphQLInvocationData invocationData, WebRequest webRequest) {
+        QueryContext queryContext = new SpringQueryContext(true, APPNAME);
```

Comment:
You are passing in APPNAME here... Soon, we'll have to understand how to authorize specific actions (say, mutating the ownership of a Dataset) against a concrete human actor (like a corpUser) at some point. To do so, we'll want to have the context of the _actual_ actor (eg. which user is making the change) to perform authorization against. 

Perhaps we can come up with some standardized "super" actor that we will always let through, and if you really cannot pass the actual actor context you can use that for your use case

This may mean that in the future you may need access to that context.